I've [removed](https://github.com/NixOS/nixpkgs/compare/8a3bf1a5e02250eca5e2b4329f50bdaa62af6cb7..2d091edabce03058078812a9cfb171b94614d122) the `pythonImportsCheck` line as it fails to initialise gtk when that's present. I'm not sure why, as it seems to use the custom `checkPhase`. Is something else trying to run the tests outside `xvfb-run`?

```
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: xdot
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/m7080pw0ryjk0jhljp55rq1hd2qy8gki-python3-3.8.6/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/nix/store/62jwz83lpwbm2sqph37fc65ji7r89xn0-python3.8-xdot-1.2/lib/python3.8/site-packages/xdot/__init__.py", line 26, in <module>
    from . import ui
  File "/nix/store/62jwz83lpwbm2sqph37fc65ji7r89xn0-python3.8-xdot-1.2/lib/python3.8/site-packages/xdot/ui/__init__.py", line 3, in <module>
    from .window import DotWidget, DotWindow
  File "/nix/store/62jwz83lpwbm2sqph37fc65ji7r89xn0-python3.8-xdot-1.2/lib/python3.8/site-packages/xdot/ui/window.py", line 40, in <module>
    from . import actions
  File "/nix/store/62jwz83lpwbm2sqph37fc65ji7r89xn0-python3.8-xdot-1.2/lib/python3.8/site-packages/xdot/ui/actions.py", line 66, in <module>
    class NullAction(DragAction):
  File "/nix/store/62jwz83lpwbm2sqph37fc65ji7r89xn0-python3.8-xdot-1.2/lib/python3.8/site-packages/xdot/ui/actions.py", line 70, in NullAction
    _tooltip_window = Gtk.Window(Gtk.WindowType.POPUP)
  File "/nix/store/8xpk06wi1f0c5907bjmw521snbiaingg-python3.8-pygobject-3.38.0/lib/python3.8/site-packages/gi/overrides/Gtk.py", line 515, in __init__
    raise RuntimeError(
RuntimeError: Gtk couldn't be initialized. Use Gtk.init_check() if you want to handle this case.
```

If it's not necessary to have `pythonImportsCheck`, this PR is ready for review.

--- a/pkgs/development/python-modules/xdot/default.nix
+++ b/pkgs/development/python-modules/xdot/default.nix
@@ -1,5 +1,5 @@
-{ lib, buildPythonPackage, fetchPypi, isPy3k
-, wrapGAppsHook, gobject-introspection, pygobject3, graphviz, gtk3 }:
+{ lib, buildPythonPackage, fetchPypi, isPy3k, python3, xvfb_run, stdenv
+, wrapGAppsHook, gobject-introspection, pygobject3, graphviz, gtk3, numpy }:
 
 buildPythonPackage rec {
   pname = "xdot";
@@ -11,9 +11,16 @@ buildPythonPackage rec {
   };
 
   disabled = !isPy3k;
-
   nativeBuildInputs = [ wrapGAppsHook ];
-  propagatedBuildInputs = [ gobject-introspection pygobject3 graphviz gtk3 ];
+  propagatedBuildInputs = [ gobject-introspection pygobject3 graphviz gtk3 numpy ];
+  checkInputs = [ xvfb_run ];
+
+  checkPhase = ''
+    xvfb-run -s '-screen 0 800x600x24' ${python3.interpreter} nix_run_setup test
+  '';
+
+  # https://github.com/NixOS/nixpkgs/pull/107872#issuecomment-752175866
+  doCheck = stdenv.isLinux;
 
   meta = with lib; {
     description = "An interactive viewer for graphs written in Graphviz's dot";

